cl/search/documents.py

```diff
@@ -1213,15 +1218,35 @@ def prepare_parties(self, instance):
             "firm_id": set(),
             "firm": set(),
         }
-        for p in instance.prefetched_parties:
-            out["party_id"].add(p.pk)
-            out["party"].add(p.name)
-            for a in p.attys_in_docket:
-                out["attorney_id"].add(a.pk)
-                out["attorney"].add(a.name)
-                for f in a.firms_in_docket:
-                    out["firm_id"].add(f.pk)
-                    out["firm"].add(f.name)
+
+        # Extract only required parties values.
+        party_values = instance.parties.values_list("pk", "name")
+        if party_values:
+            out["party_id"].update([pk for pk, _ in party_values])
+            out["party"].update([name for _, name in party_values])
```

Comment:
I forget if you can use iterator on a `values_list` query, but even if you can't, I suspect doing this in a single loop should be faster, simpler and use less memory:

```
for pk, name in party_values.iterator():
    out["party_id"].add(pk)
    out["party"].add(name)
```

Reply:
Yes, `iterator()` works on a value_list; it must be better in terms of memory usage. Changed. Thanks for the suggestion!